pkg/handler/collector/oci/oci.go

```diff
@@ -183,89 +208,192 @@ func (o *ociCollector) getTagsAndFetch(ctx context.Context, repo string, tags []
 // A workaround for this would be to run the collector again with a specific tag without polling and ingest like normal
 func (o *ociCollector) fetchOCIArtifacts(ctx context.Context, repo string, rc *regclient.RegClient, image ref.Ref, docChannel chan<- *processor.Document) error {
 	logger := logging.FromContext(ctx)
-
 	// attempt to request only the headers, avoids Docker Hub rate limits
 	m, err := rc.ManifestHead(ctx, image)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed retrieving manifest head: %w", err)
```

Comment:
The function `fetchOCIArtifacts` is quite long and does quite a few different things, which makes it harder to test, debug, and maintain. 

Consider breaking it down into smaller, more manageable functions.

Reply:
Agreed, it felt like that to me as I was adding to it. Let me see if I can improve it.

Reply:
Broke it down into more readable functions with one purpose. Let me know if you agree with the direction.